scripts/entrypoint.sh

```diff
@@ -93,11 +93,17 @@ log_in_json info "Copying CNI plugin binary and config file ... "
 
 install aws-cni "$HOST_CNI_BIN_PATH"
 
-sed -i s~__VETHPREFIX__~"${AWS_VPC_K8S_CNI_VETHPREFIX}"~g 10-aws.conflist
-sed -i s~__MTU__~"${AWS_VPC_ENI_MTU}"~g 10-aws.conflist
-sed -i s~__PLUGINLOGFILE__~"${AWS_VPC_K8S_PLUGIN_LOG_FILE}"~g 10-aws.conflist
-sed -i s~__PLUGINLOGLEVEL__~"${AWS_VPC_K8S_PLUGIN_LOG_LEVEL}"~g 10-aws.conflist
-cp 10-aws.conflist "$HOST_CNI_CONFDIR_PATH"
+# create a temporary file to be manipulated by sed (ergh).
+# this enables you to mount your own config file from a configmap that is read only
+temp_config_file=$(mktemp)
+cp 10-aws.conflist "$temp_config_file"
```

Comment:
I agree the original code was already leading that way, but `cp` then `sed`-in-place is a bit weird given that sed is already a fine "stream editor" :P

May I suggest:
```
temp_config_file="$HOST_CNI_CONFDIR_PATH/10-aws.conflist.tmp"
 
sed \
  -e s~__VETHPREFIX__~"${AWS_VPC_K8S_CNI_VETHPREFIX}"~g \
  -e s~__MTU__~"${AWS_VPC_ENI_MTU}"~g \
  -e s~__PLUGINLOGFILE__~"${AWS_VPC_K8S_PLUGIN_LOG_FILE}"~g \
  -e s~__PLUGINLOGLEVEL__~"${AWS_VPC_K8S_PLUGIN_LOG_LEVEL}"~g \
  10-aws.conflist > "$temp_config_file"

mv "$temp_config_file" "$HOST_CNI_CONFDIR_PATH/10-aws.conflist"
```

In addition, the above code writes the tempfile to the same filesystem as the final file, and then atomically renames it into place.

Reply:
Thanks for the review! I thought about something like this, but didn't want to change too much, and hadn't read the cni code to ensure it only reads *.conflist. I just did and it does, so this approach is much nicer!

Reply:
I have removed the tempfile madness and used it as a "stream editor" and redirected the contents into the destination file. I think this makes more sense!

Reply:
It's much shorter/cleaner, I agree :)

My concern is that now it might fail midway (or race with kubelet logic) and the kubelet will end up reading half the file.  .. Now that I think about it, I think that's ok because it's json and it won't be valid until the last closing brace is seen anyway - the worst that will happen is that the kubelet will try to start some pods that will fail-retry a few times until the CNI pod is restarted and fixes the config.